Linux - Workouts 1 - 3 Improvements (#2096)
enkidevs/curriculum: linux/user-and-file-management/linux-intro/linux-common-terminology.md

---
author: kapnobatai136

aspects:
  - introduction

type: normal

category: must-know

tags:
  - introduction
  - linux

---

# Common Terminology

---
## Content

Let's go over a few terms that you'll see a lot in this course.

> Linux comes with an improved version of the shell, which is called **bash**[1]. From now on, we'll be using these terms interchangeably.

Reviewer: bash also refers to the programming language used to interpret the code within the shell
wondering if we should make this mention

Reviewer 2: Hmmm, I feel like it might be too confusing this early on?

Author: I'll add a note like:

> 💡 Fun fact: **bash** also refers to the programming language used to interpret the code within the shell.

---
author: kapnobatai136

aspects:
  - introduction

type: normal

category: must-know

tags:
  - introduction
  - linux

---

# Common Terminology

---
## Content

Let's go over a few terms that you'll see a lot in this course.

A **command** is just a piece of text that you type into the **terminal**. Think of this as an instruction that you give to your computer.

Sometimes you might type some commands which are restricted. That's where the **root user** comes in. It's the unique admin user, with superpowers. Unsurprisingly, it's also called the **superuser**.

The **package manager** removes the tedious process of installing or uninstalling apps. Think of a **package** as being an app.

> Remember the terminal? Well, this is also referred to as the **shell**. 

Reviewer 2: The terminal is mentioned a few sentences above so "remember" seems unnecessary?

Also, a shell isn't really the same as the terminal. A shell is a program running in the terminal.

Reviewer 2: Maybe this part about the shell can be:

```suggestion
The main part of the terminal that reads the commands and executes them is called a **shell**
```

> Linux comes with an improved version of the shell, which is called **bash**[1] (**Bourne-Again SHell**). From now on, we'll be using these terms interchangeably.

Reviewer 2: Should we mention the "Bourne-Again SHell" part here if we also mention it in the footnote?

Author: no, forgot to remove it.

> **Note:** as with all the skills in the app, if you want to access more advanced or specific content at any time, **feel free to use the *Skill Map* tab** to jump to the workouts most relevant for you.

---
## Practice

> Commands are typed in the ???.
> 
> * terminal
> * package manager
> * superuser manager

Reviewer 2: See: https://superuser.com/a/144668

```suggestion
Commands are typed into the ??? and processed by the ???.

* terminal
* shell
* package manager
* superuser manager
```

---
## Revision

The **root user** is also called the ???.

* superuser
* extrauser
* super
* best-user

---
## Footnotes

[1:Bash]

> The name is an acronym for the ‘Bourne-Again SHell’, a pun on Stephen Bourne, the author of the direct ancestor of the current Unix shell `sh`, which appeared in the Seventh Edition Bell Labs Research version of Unix.

Reviewer 2: The quotes are wrong:

```suggestion
The name is an acronym for the 'Bourne-Again SHell', a pun on Stephen Bourne, the author of the direct ancestor of the current Unix shell `sh`, which appeared in the Seventh Edition Bell Labs Research version of Unix.
```

Reviewer 2: Also, we suddenly introduce Unix here without ever mentioning it before. This would seem like a typo to most users unfamiliar with it (and since we're in the Linux topic)

```suggestion
The name is an acronym for the ‘Bourne-Again SHell’, a pun on Stephen Bourne, the author of the direct ancestor of the another shell called `sh`.
```
